deps/build.jl

```diff
@@ -149,10 +150,41 @@ pm_cxx = chomp(read(`$perl $pm_config --cc`, String))
 jlcxx_cmake_dir = joinpath(CxxWrap.prefix_path(), "lib", "cmake", "JlCxx")
 julia_exec = joinpath(Sys.BINDIR , "julia")
 
+xcode_typeinfo_bug = false
+
+if (Sys.isapple())
+   # Lets have some fun for xcode 11.4 until
+   # https://github.com/llvm/llvm-project/commit/2464d8135e
+   # arrives.
+   #
+   # We build a jlcxx library that uses std::string which will should
+   # abort with a failed assertion has_julia_type<T> if we are building
+   # with xcode 11.4 but libcxxwrap-julia was built with an older libc++.
+   #
+   # Read the above commit message some details; the effect of merged
+   # vs non-merged type_info is that for the former memory addresses are
+   # used as hash_code(), for the latter the type_info.name() string is
+   # hashed.
+
+   cd(joinpath(@__DIR__, "xcodetypeinfo"))
+   run(`$(CMake.cmake) -DJulia_EXECUTABLE=$julia_exec -DJlCxx_DIR=$jlcxx_cmake_dir .`)
+   run(`make -j1`)
+   libpath = joinpath(@__DIR__, "xcodetypeinfo", "libhello.$dlext")
+   res = run(pipeline(Cmd(`$julia_exec --project -e "using CxxWrap; @wrapmodule(\"$libpath\", :define_module_hello); @initcxx;"`,ignorestatus=true),stdout=devnull,stderr=devnull))
```

Comment:
Should this perhaps use `Base.julia_cmd()`?
```suggestion
   res = run(pipeline(Cmd(`$(Base.julia_cmd()) --project -e "using CxxWrap; @wrapmodule(\"$libpath\", :define_module_hello); @initcxx;"`,ignorestatus=true),stdout=devnull,stderr=devnull))
```